cordapp-example/contracts-kotlin/src/main/kotlin/com/example/contract/IOUContract.kt

```diff
@@ -30,24 +30,35 @@ class IOUContract : Contract {
      * considered valid.
      */
     override fun verify(tx: LedgerTransaction) {
-        val command = tx.commands.requireSingleCommand<Commands.Create>()
-        requireThat {
-            // Generic constraints around the IOU transaction.
-            "No inputs should be consumed when issuing an IOU." using (tx.inputs.isEmpty())
-            "Only one output state should be created." using (tx.outputs.size == 1)
-            val out = tx.outputsOfType<IOUState>().single()
-            "The lender and the borrower cannot be the same entity." using (out.lender != out.borrower)
-            "All of the participants must be signers." using (command.signers.containsAll(out.participants.map { it.owningKey }))
-
-            // IOU-specific constraints.
-            "The IOU's value must be non-negative." using (out.value > 0)
+        val command = tx.commands.requireSingleCommand<Commands>()
+
+        when (command.value) {
+            is Commands.Create -> requireThat {
+                // Generic constraints around the IOU transaction.
+                "No inputs should be consumed when issuing an IOU." using (tx.inputs.isEmpty())
+                //Only one output state should be created
+                val out = tx.outputsOfType<IOUState>().single()
```

Comment:
Better check the size first so that you can give a nice error message instead of the default one that happens with `single`.